node/src/block_ranges.rs

```diff
@@ -443,6 +457,33 @@ impl Sub<&BlockRanges> for BlockRanges {
     }
 }
 
+impl Sub<RangeInclusive<u64>> for BlockRanges {
+    type Output = Self;
+
+    fn sub(mut self, range: RangeInclusive<u64>) -> Self::Output {
+        if !range.is_empty() {
+            self.remove_relaxed(range).expect("shouldn't panic");
+        }
+        self
+    }
+}
+
+impl SubAssign<RangeInclusive<u64>> for BlockRanges {
+    fn sub_assign(&mut self, range: RangeInclusive<u64>) {
+        if !range.is_empty() {
+            self.remove_relaxed(range).expect("shouldn't panic");
+        }
+    }
+}
+
+impl Iterator for BlockRanges {
```

Comment:
Implement [DoubleEndedIterator](https://doc.rust-lang.org/std/iter/trait.DoubleEndedIterator.html) too. With that we will be able to use `rev()`.

Reply:
implemented :+1: